packages/agreement-process/src/model/domain/errors.ts

```diff
@@ -102,11 +109,85 @@ export function agreementNotInExpectedState(
   });
 }
 
-export function tenantIdNotFound(tenantId: string): ApiError {
+export function tenantIdNotFound(
+  httpStatus: number,
+  tenantId: string
+): ApiError {
   return new ApiError({
     detail: `Tenant ${tenantId} not found`,
     code: errorCodes.tenantIdNotFound,
-    httpStatus: 404,
+    httpStatus,
     title: "Tenant not found",
   });
 }
+
+export function agreementSubmissionFailed(agreementId: string): ApiError {
+  return new ApiError({
+    detail: `Unable to activate agreement ${agreementId}. Please check if attributes requirements and suspension flags are satisfied`,
+    code: errorCodes.agreementSubmissionFailed,
+    httpStatus: 400,
+    title: "Unable to activate agreement $agreementId",
+  });
+}
+
+export function consumerWithNotValidEmail(
+  agreementId: string,
+  tenantId: string
+): ApiError {
+  return new ApiError({
+    detail: `Agreement ${agreementId} has a consumer tenant ${tenantId} with no valid email`,
+    code: errorCodes.consumerWithNotValidEmail,
+    httpStatus: 400,
+    title: "Agreement with invalid consumer email",
+  });
+}
+
+export function agreementStampNotFound(stamp: string): ApiError {
+  return new ApiError({
+    detail: `Agreement stamp ${stamp} not found`,
+    code: errorCodes.agreementStampNotFound,
+    httpStatus: 500,
+    title: "Stamp not found",
+  });
+}
+
+export function agreementMissingUserInfo(userId: string): ApiError {
+  return new ApiError({
+    detail: `Some mandatory info are missing for user ${userId}`,
+    code: errorCodes.agreementMissingUserInfo,
+    httpStatus: 500,
+    title: "Some mandatory info are missing for user",
+  });
+}
+
+export function agreementSelfcareIdNotFound(tenantId: string): ApiError {
+  return new ApiError({
+    detail: `Selfcare id not found for tenant ${tenantId}`,
+    code: errorCodes.agreementSelfcareIdNotFound,
+    httpStatus: 500,
+    title: "Selfcare id not found for tenant",
+  });
+}
+
+export function agreementDescriptorNotFound(
+  eserviceId: string,
+  descriptorId: string
+): ApiError {
+  return new ApiError({
+    detail: `Descriptor ${descriptorId} not found in EService ${eserviceId}`,
+    code: errorCodes.agreementDescriptorNotFound,
+    httpStatus: 500,
+    title: "Descriptor not found",
+  });
+}
+
+export function agreementSubmitOperationNotAllowed(
```

Comment:
in error refactor PR #118 we have `operationForbidden` error instead of `operationNotAllowed`, if not necessary to distinguish those two errors I could proceed to use only `operationForbidden` and replace `operationNotAllowed` error PR #118. 
https://github.com/pagopa/interop-be-monorepo/blob/84757b749cb0da8d8a55c108893c92dbbe7b64f6/packages/models/src/errors.ts#L138

Reply:
yes they can be considered the same error, we can use the one in commons after the PR 118 👍